### ISSUE TITLE
Certain attribute name and value combinations crash the VS Code extension

### ISSUE THREAD

**Issue by jozefmery** [unocss/unocss#2206]
### UnoCSS version

Started in `v0.49.5`, present also in `v0.49.6`.

### Describe the bug

Using attributify mode, certain (very specific) attribute name and value combinations crash the VS Code extension with the following message:
```
⚠️ Error on annotation
TypeError: Cannot read properties of undefined (reading 'match')
```
Based on the change-log and the error message, my best guess is that the issue is caused somewhere in #2186.

Generated CSS seems to be correct.

### Reproduction

The minimal reproduction I was able to find is the following: `<div p="1" gap="md:1" />`. 
See [playground](https://uno.antfu.me/play/?html=DwEwlgbgBADgvAIgIwKgcwIbwQWxALhSgHoA%2BIA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQ61AA9IsFBgCGbADbw0mHPkLAiACgTq45SjRhUAXHADaT5y4EYYQg7AEoSP2d%2BNwYmVg4ubjCI5wBdCLFQ9SA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwBoQJ84AHJeASwDd9NIBXIkAd2oBMwALTAdgAcARmI981AOY8wAgEwA2AL5A) reproduction, where the problem is observable by the missing underline. Minimal changes, e.g. removing either attribute, or removing the `md` variant, or changing `p="1"` to `p="2"`, or `gap="md:1"` to `gap="md:2"` wierdly resolve the issue. 

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by zyyv** [comment]
We don't have this rule `gap="md:1"`, the correct class name is `md="gap-1"` or `md-gap="1"`, the error may be caused by wrong rule.

**Comment by jozefmery** [comment]
I'm not so sure the rule is wrong. Notice that if you change `gap="md:1"` to `gap="md:2"` everything works. Furthermore, the generated CSS is correct in both cases.

**Comment by zyyv** [comment]
You can refer to the [interactive docs](https://uno.antfu.me/?s=gap-md-1).

And you should follow the hermit writing rules for atomic classes in UnoCSS.  `variant-rule`

**Comment by sibbng** [comment]
I confirm this is a bug. Should be fixed when #2207 is merged.
